src/util/util_env.cpp

```diff
@@ -85,6 +89,17 @@ namespace dxvk::env {
     size_t count = readlink("/proc/self/exe", exePath.data(), exePath.size());
 
     return std::string(exePath.begin(), exePath.begin() + count);
+#elif defined(__FreeBSD__)
+    int mib[4] = {CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, getpid()};
+    std::string exePath(PATH_MAX, '\0');
+    size_t size = exePath.size();
+
+    if (sysctl(mib, 4, exePath.data(), &size, NULL, 0) != 0) {
```

Comment:
It would be cool, but you can't use std::string like this, sorry!

Make a char array on the stack then put that into a std::string.